docs/resources/client/models/items.md

```diff
@@ -0,0 +1,1113 @@
+                // For when tintindex: 0
+                new Constant(
+                    // Pure green
+                    0x00FF00
+                ),
+                // For when tintindex: 1
+                new Dye(
+                    // Pure blue
+                    // Only is called if `DataComponents#DYED_COLOR` is not set
+                    0x0000FF
+                )
+            )
+        )
+    );
+}
+```
+
+</TabItem>
+</Tabs>
+
+Creating your own `ItemTintSource` is similar to any other codec-based registry object. You make a class that implements `ItemTintSource`, create a `MapCodec` to encode and decode the object, and register the codec to its registry via `RegisterColorHandlersEvent.ItemTintSources` on the [mod event bus][modbus]. The `ItemTintSource` only contains one method `calculate`, which takes in the current `ItemStack`, the level the stack is in, and the entity holding the stack to return an opaque color in ARGB format, where the top 8 bits are 0xFF.
+
+```java
+public record DamageBar(int defaultColor) implements ItemTintSource {
+
+    // The map codec to register
+    public static final MapCodec<DamageBar> MAP_CODEC = ExtraCodecs.RGB_COLOR_CODEC.fieldOf("default")
+        .xmap(DamageBar::new, DamageBar::defaultColor);
+
+    public DamageBar(int defaultColor) {
+        // Make sure the passed in color is opaque
+        this.defaultColor = ARGB.opaque(defaultColor);
+    }
+
+    @Override
+    public int calculate(ItemStack stack, @Nullable ClientLevel level, @Nullable LivingEntity entity) {
+        return stack.isDamaged() ? ARGB.opaque(stack.getBarColor()) : defaultColor;
+    }
+
+    @Override
+    public MapCodec<DamageBar> type() {
+        return MAP_CODEC;
+    }
+}
+
+// In some class where the event is registered to the mod event bus
+@SubscribeEvent
+public static void registerItemTintSources(RegisterColorHandlersEvent.ItemTintSources event) {
+    event.register(
+        // The name to reference as the type
+        ResourceLocation.fromNamespaceAndPath("examplemod", "damage_bar"),
+        // The map codec
+        DamageBar.MAP_CODEC
+    )
+}
+```
+
+```json5
+// For some item 'examplemod:example_item'
+// JSON at 'assets/examplemod/items/example_item.json'
+{
+    "model": {
+        "type": "minecraft:model",
+        // Points to 'assets/examplemod/models/item/example_item.json'
+        "model": "examplemod:item/example_item",
+        // A list of tints to apply
+        "tints": [
+            {
+                // For when tintindex: 0
+                "type": "examplemod:damage_bar",
+                // 0x00FF00 (or pure green)
+                "default": 65280
+            }
+        ]
+    }
+}
+```
+
+## Composite Models
+
+SSometimes, you may want to register multiple models for a single item. While this can be done directly with the [composite model loader][composite], for item models, there is a custom `minecraft:composite` type, which takes a list of models to render.
```

Comment:
```suggestion
Sometimes, you may want to register multiple models for a single item. While this can be done directly with the [composite model loader][composite], for item models, there is a custom `minecraft:composite` type which takes a list of models to render.
```